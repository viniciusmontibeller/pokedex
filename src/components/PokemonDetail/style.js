import styled from 'styled-components'

const Container = styled.div`
    display: flex;
    flex-direction: column;
    gap: 30px;
    align-items: center;
`

const DetailContainer = styled.div`
    background-color: ${props => props.theme.defaultValues.transparentBackground};
    box-shadow: ${props => props.theme.defaultValues.darkGlassShadow};
    backdrop-filter: ${props => props.theme.defaultValues.backgroundFilter};
    padding: 30px;
    border-radius: 20px;
    display: flex;
    max-width: 1000px;
    gap: 20px;
    letter-spacing: 1px;
    color: ${props => props.theme.theme.color};

    h1, h2, h3{
        text-transform: capitalize;
    }

    h2 {
        margin-bottom: 15px;
    }

    @media(max-width: 800px){
        flex-direction: column;
        align-items: center;
    }

    @media(min-width: 500px){
        gap: 50px;
    }
`

const Name = styled.div`
    display: flex;
    align-items: baseline;
    justify-content: space-between;
    box-shadow: ${props => props.theme.defaultValues.glassShadow};
    border-radius: 10px;
    padding: 5px 10px;
    p{
        font-weight: 500;
        font-size: 25px;
        opacity: 0.7;
    }

    @media(max-width: 500px){
        h1, p {
            font-size: 20px;
        }
    }
`

const Moves = styled.ul`
      padding: 15px 10px;
      max-height: 200px;
      overflow-y: scroll;
      max-width: 400px;
      display: flex;
      gap: 10px;
      flex-wrap: wrap;
      justify-content: center;
      border-radius: 10px;
      box-shadow: hsla(211, 76%, 52%, 0.3) 3px 3px 6px 0px inset, hsla(211, 76%, 70%, 0.3) -3px -3px 6px 1px inset;
    
    ::-webkit-scrollbar {
    width: 8px;
    }

    ::-webkit-scrollbar-track {
    box-shadow: hsla(211, 76%, 52%, 0.3) 3px 3px 1px 0px inset, hsla(211, 76%, 70%, 0.3) -3px -3px 1px 1px inset;
    border-radius: 10px;
    }

    ::-webkit-scrollbar-thumb {
    background-color: ${props => props.theme.pokemonTypeColor[props.colorType]};
    border-radius: 10px;
    }

      li {
        padding: 5px 20px;
        background-color: ${props => props.theme.pokemonTypeColor[props.colorType]};
        font-weight: 500;
        border-radius: 40px;
      }
   `
const Ability = styled.ul`
    display: flex;
    flex-direction: column;
    gap: 20px;

    li{
        border-radius: 5px;
        box-shadow: hsla(211, 76%, 52%, 0.3) 3px 3px 6px 0px inset, hsla(211, 76%, 70%, 0.3) -3px -3px 6px 1px inset; 
        
        h3, p {
        padding: 2px 10px;
        }

        p{
            border-bottom-left-radius: inherit;
            border-bottom-right-radius: inherit;
        }

        h3{
        border-top-left-radius: inherit;
        border-top-right-radius: inherit;
        box-shadow: ${props => props.theme.defaultValues.shadow};
        }
    }
`

const Info= styled.div`
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    max-width: 450px;
    flex: 1;
    img{
        padding: 40px;
        max-width: 100%;
        filter: drop-shadow( 0px 0px 10px white);
    }
`

const Description = styled.div`
    display: flex;
    flex-wrap: wrap;
    flex: 1;
    gap: 20px;
    max-width: 400px;
    justify-content: center;
    @media(max-width: 500px){
        h2{
            font-size: 1.3rem;
        }
        h3{
            font-size: 1rem;
        }
        p,li{
            font-size: 0.8rem;
        }
        
    }
`

const AbilitiesContainer = styled.div`
    max-width: 400px;
    h2{
        text-align: center;
    }
`

const MovesContainer = styled.div`
    h2{
        text-align: center;
    }
`

export { Container, DetailContainer, Moves, Ability, Info, Description, MovesContainer, Name, AbilitiesContainer }