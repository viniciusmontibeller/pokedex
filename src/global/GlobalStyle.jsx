import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
    *{
        margin: 0;
        padding: 0;
        box-sizing: border-box;
    }

    ul {
        list-style: none;
    }

    a {
        text-decoration: none;
    }

    body{
        background: url(${props => props.theme.theme.backgroundImage}) no-repeat center center fixed; 
        background-size: cover;
        background-color: ${props => props.theme.theme.backgroundColor};
        font-family: ${props => props.theme.defaultValues.fontFamily};

        #root {
            height: auto;
        }
    }
`

export { GlobalStyle }