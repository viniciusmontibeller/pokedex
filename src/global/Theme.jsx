import React, { useEffect } from 'react'
import { ThemeProvider } from 'styled-components'
import { useState } from 'react'
import LightBackground from '../assets/lightbg.jpg'
import DarkBackground from '../assets/darkbg.jpg'

const themes = {
    light: {
        color: "rgba(34, 34, 34, 1)",
        backgroundColor: "rgba(255, 255, 255, 1)",
        backgroundColorHover: "rgba(34, 34, 34, 0.2)",
        backgroundImage: LightBackground,
    },
    dark: {
        color: "rgba(255, 255, 255, 1)",
        backgroundColor: "rgba(34, 34, 34, 1)",
        backgroundColorHover: "rgba(255, 255, 255, 0.3)",
        backgroundImage: DarkBackground,
    }
};

const defaultValues = {
    transparentBackground: "rgba(255, 255, 255, 0.1)",
    glassShadow: "hsla(211, 76%, 52%, 0.3) 3px 3px 6px 0px inset, hsla(211, 76%, 70%, 0.3) -3px -3px 6px 1px inset, 0 0 10px 2px rgba(0, 0, 0, 0.35)",
    darkGlassShadow: "hsla(211, 76%, 52%, 0.3) 3px 3px 6px 0px inset, hsla(211, 76%, 70%, 0.3) -3px -3px 6px 1px inset, 0 0 10px 5px rgba(0, 0, 0, 0.35)",
    shadow: "0 0 10px 1px rgba(0, 0, 0, 0.35)",
    hoverShadow: "0 0 10px 5px rgba(0, 0, 0, 0.35)",
    transition: "0.2s ease-in-out",
    backgroundFilter: "blur(8px)",
    fontFamily: "'Nunito', sans-serif"
}

const pokemonTypeColor = {
    fire: "#EE8130",
    grass: "#7AC74C",
    electric: "#F7D02C",
    water: "#6390F0",
    ground: "#E2BF65",
    rock: "#B6A136",
    fairy: "#D685AD",
    poison: "#A33EA1",
    bug: "#A6B91A",
    dragon: "#6F35FC",
    psychic: "#F95587",
    flying: "#A98FF3",
    fighting: "#C22E28",
    normal: "#A8A77A",
    ice: "#96D9D6",
    ghost: "#735797",
    dark: "#705746",
    steel: "#B7B7CE",
}

const localTheme = JSON.parse(localStorage.getItem('theme')) || themes.light

const Theme = ({children}) => {
        const [ theme, setTheme ] = useState(localTheme)

        useEffect(() => {
            localStorage.setItem('theme', JSON.stringify(theme))
        }, [theme])

        return (
        <ThemeProvider theme={{theme, setTheme, pokemonTypeColor, defaultValues}}>
            {children}
        </ThemeProvider>
        )
}

export { Theme, themes }