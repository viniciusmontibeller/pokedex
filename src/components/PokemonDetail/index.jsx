import { DetailContainer, Moves, Ability, Info, Description, MovesContainer, Name, AbilitiesContainer } from './style'
import { Types, Type } from '../Types'


const PokemonDetail = ({ name, id, image, types, abilities, moves }) => {

   return (
      <DetailContainer>
         <Info colorType={types?.[0].type.name}>
            <Name>
               <h1>{name}</h1>
               <p># {id}</p>
            </Name>
            <img src={image} alt={name} />
            <Types>
               {
                  types?.map((type, index) => {
                     return <Type key={index} biggerType type={type.type.name}>
                        {type.type.name}
                     </Type>
                  })
               }
            </Types>
         </Info>
         <Description>
            <AbilitiesContainer>
               <h2>Abilities</h2>
               <Ability colorType={types?.[0].type.name}>
                  {
                     abilities.map((ability, index) => {
                        return (<li key={index}>
                           <h3>{ability.name}</h3>
                           {ability.effect_entries.map((effect, index) => {
                              if (effect.language.name.includes("en")) {
                                 return <p key={index}>{effect.short_effect}</p>
                              }
                           })}
                        </li>
                        )
                     })
                  }
               </Ability>
            </AbilitiesContainer>
            <MovesContainer>
               <h2>Moves</h2>
               <Moves colorType={types?.[0].type.name}>
                  {
                     moves?.map((move, index) => {
                        return <li key={index}>{move.move.name}</li>
                     })
                  }
               </Moves>
            </MovesContainer>
         </Description>
      </DetailContainer>
   )
}

export { PokemonDetail }