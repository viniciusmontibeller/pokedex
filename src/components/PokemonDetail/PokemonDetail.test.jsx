import { PokemonDetail } from "./PokemonDetail";
import { render } from "@testing-library/react";
import { Theme } from "../../global/Theme";

describe("Pokemon detail component", () => {
    const pokemonFake = {
        id: '6',
        name: 'charizard',
        image: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/6.png',
        types:
            [
                {
                    "slot": 1,
                    "type": {
                        "name": "grass",
                        "url": "https://pokeapi.co/api/v2/type/12/"
                    }
                },
                {
                    "slot": 2,
                    "type": {
                        "name": "poison",
                        "url": "https://pokeapi.co/api/v2/type/4/"
                    }
                }
            ],
        abilities: [{
            effect_entries:
                [
                    {
                        "effect": "Wenn ein Pokémon mit dieser Fähigkeit nur noch 1/3 seiner maximalen hp oder weniger hat, werden all seine fire Attacken verstärkt, so dass sie 1,5× so viel regular damage anrichten wie sonst.",
                        "language": {
                            "name": "de",
                            "url": "https://pokeapi.co/api/v2/language/6/"
                        },
                        "short_effect": "Erhöht den Schaden von fire Attacken um 50% wenn nur noch 1/3 der maximalen hp oder weniger übrig sind."
                    },
                    {
                        "effect": "When this Pokémon has 1/3 or less of its HP remaining, its fire-type moves inflict 1.5× as much regular damage.",
                        "language": {
                            "name": "en",
                            "url": "https://pokeapi.co/api/v2/language/9/"
                        },
                        "short_effect": "Strengthens fire moves to inflict 1.5× damage at 1/3 max HP or less."
                    }
                ]
        },
        {
            effect_entries:
                [
                    {
                        "effect": "Wenn ein Pokémon mit dieser Fähigkeit nur noch 1/3 seiner maximalen hp oder weniger hat, werden all seine fire Attacken verstärkt, so dass sie 1,5× so viel regular damage anrichten wie sonst.",
                        "language": {
                            "name": "de",
                            "url": "https://pokeapi.co/api/v2/language/6/"
                        },
                        "short_effect": "Erhöht den Schaden von fire Attacken um 50% wenn nur noch 1/3 der maximalen hp oder weniger übrig sind."
                    },
                    {
                        "effect": "When this Pokémon has 1/3 or less of its HP remaining, its fire-type moves inflict 1.5× as much regular damage.",
                        "language": {
                            "name": "en",
                            "url": "https://pokeapi.co/api/v2/language/9/"
                        },
                        "short_effect": "Strengthens fire moves to inflict 1.5× damage at 1/3 max HP or less."
                    }
                ]
        }
        ],
        moves: [
            {
                move: {
                    name: "mega-punch"
                }
            },
            {
                move: {
                    name: "mega-punch"
                }
            },
            {
                move: {
                    name: "mega-punch"
                }
            },
            {
                move: {
                    name: "mega-punch"
                }
            },
            {
                move: {
                    name: "mega-punch"
                }
            }
        ]

    }
    it("should render a pokemon's details", () => {
        const {asFragment} = render(<Theme><PokemonDetail {...pokemonFake} /></Theme>)
        expect(asFragment).toMatchSnapshot()
    })
})
