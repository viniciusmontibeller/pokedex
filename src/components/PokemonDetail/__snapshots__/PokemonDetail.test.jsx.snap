// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon detail component should render a pokemon's details 1`] = `[Function]`;
